arcgis-ios-sdk-samples/Display information/Set up location-driven Geotriggers/SetUpLocationDrivenGeotriggersViewController.swift

```diff
@@ -0,0 +1,268 @@
+    var nearbyFeatures: [String: AGSArcGISFeature] = [:]
+    /// An array of observers for geotrigger monitor notifications.
+    var observers: [NSObjectProtocol] = []
+    /// An array of geotrigger monitors.
+    var geotriggerMonitors: [AGSGeotriggerMonitor] = []
+    
+    /// The name of the current garden section feature. If currently not in any
+    /// garden section, it will be `nil`.
+    var currentSectionName: String? {
+        featureNamesInFenceGeotrigger[sectionFenceGeotriggerName]?.last
+    }
+    /// The names of nearby point-of-interest features.
+    var nearbyPOINames: [String] {
+        featureNamesInFenceGeotrigger[poiFenceGeotriggerName] ?? []
+    }
+    
+    // MARK: Methods
+    
+    /// Create a simulated location data source from a GeoJSON.
+    func makeDataSource(polylineJSONString: String) -> AGSSimulatedLocationDataSource {
+        let simulatedDataSource = AGSSimulatedLocationDataSource()
+        let jsonObject = try? JSONSerialization.jsonObject(with: polylineJSONString.data(using: .utf8)!)
+        let routePolyline = try? AGSPolyline.fromJSON(jsonObject!) as? AGSPolyline
+        // Densify the polyline to control the simulation speed.
+        let densifiedRoute = AGSGeometryEngine.geodeticDensifyGeometry(
+            routePolyline!,
+            maxSegmentLength: 50.0,
+            lengthUnit: .meters(),
+            curveType: .geodesic
+        ) as! AGSPolyline
+        simulatedDataSource.setLocationsWith(densifiedRoute)
+        return simulatedDataSource
+    }
+    
+    /// Create a geotrigger monitor and observe its notifications.
+    ///
+    /// - Parameters:
+    ///   - feed: The `AGSGeotriggerFeed` that is monitored for changes.
+    ///   - featureTable: The `AGSFeatureTable` that contains the features to use
+    ///    in the `AGSFeatureFenceParameters`.
+    ///   - bufferDistance: A buffer distance in meters to apply to the features
+    ///   when checking if an `AGSFenceGeotrigger` condition is met.
+    ///   - fenceGeotriggerName: The name for the `AGSFenceGeotrigger`.
+    func createGeotriggerMonitor(feed: AGSGeotriggerFeed, featureTable: AGSServiceFeatureTable, bufferDistance: Double, fenceGeotriggerName: String) {
+        let fenceParameters = AGSFeatureFenceParameters(featureTable: featureTable, bufferDistance: bufferDistance)
+        // The Arcade expression in the fence geotrigger returns the value for
+        // the "name" field of the feature that triggered the monitor.
+        let fenceGeotrigger = AGSFenceGeotrigger(
+            feed: feed,
+            ruleType: .enterOrExit,
+            fenceParameters: fenceParameters,
+            messageExpression: AGSArcadeExpression(expression: "$fenceFeature.name"),
+            name: fenceGeotriggerName
+        )
+        
+        // Create and start the geotrigger monitor.
+        let geotriggerMonitor = AGSGeotriggerMonitor(geotrigger: fenceGeotrigger)
+        geotriggerMonitor.start { _ in }
```

Comment:
I checked the SDK src and it seems this `start(completion:)` block is not nullable or has a default value.

Reply:
Could you please create an issue for this?

Reply:
Sure. Issue at `runtime/cocoa#10760`